controllers/fenceagentsremediation_controller_test.go

```diff
@@ -123,44 +123,45 @@ var _ = Describe("FAR Controller", func() {
 			})
 		})
 	})
+
 	Context("Reconcile", func() {
-		nodeKey := client.ObjectKey{Name: workerNode}
-		farNamespacedName := client.ObjectKey{Name: workerNode, Namespace: defaultNamespace}
 		farNoExecuteTaint := utils.CreateFARNoExecuteTaint()
 		conditionStatusPointer := func(status metav1.ConditionStatus) *metav1.ConditionStatus { return &status }
+
 		BeforeEach(func() {
 			// Create two VAs and two pods, and at the end clean them up with DeferCleanup
 			va1 := createVA(vaName1, workerNode)
 			va2 := createVA(vaName2, workerNode)
 			testPod := createRunningPod("far-test-1", testPodName, workerNode)
 			DeferCleanup(cleanupTestedResources, va1, va2, testPod)
+
 			farPod := createRunningPod("far-manager-test", farPodName, "")
 			DeferCleanup(k8sClient.Delete, context.Background(), farPod)
 		})
+
 		JustBeforeEach(func() {
 			// Create node, and FAR CR, and at the end clean them up with DeferCleanup
 			Expect(k8sClient.Create(context.Background(), node)).To(Succeed())
 			DeferCleanup(k8sClient.Delete, context.Background(), node)
+
 			Expect(k8sClient.Create(context.Background(), underTestFAR)).To(Succeed())
-			DeferCleanup(k8sClient.Delete, context.Background(), underTestFAR)
+			DeferCleanup(cleanupFar(), context.Background(), underTestFAR)
```

Comment:
Is this function really needed?

Reply:
In my tests the `JustBeforeEach` step always finds the previously created CR still available in the cluster and fails. Like the following

```
[JustBeforeEach] /home/clobrano/workspace/medik8s/upstream/fence-agents-remediation/rework-fa-async1/controllers/fenceagentsremediation_controller_test.go:142                                          
    190   [It] /home/clobrano/workspace/medik8s/upstream/fence-agents-remediation/rework-fa-async1/controllers/fenceagentsremediation_controller_test.go:231                                                      
      1                                                                                                                                                                                                           
      2   [FAILED] Expected success, but got an error:                                                                                                                                                            
      3       <*errors.StatusError | 0xc0004c4280>:                                                                                                                                                               
      4       object is being deleted: fenceagentsremediations.fence-agents-remediation.medik8s.io "worker-0" already exists
```
This is why `cleanupFar()` waits for the CR deletion.
I noticed though that the comment is wrong, it doesn't remove FAR finalizer. I'll update that

Reply:
So how about just delete far with force (zero grace period)?

Reply:
I forgot that `cleanupFar` also ignores when FAR CR does not exist already, which happens in one of the new tests.

> So how about just delete far with force (zero grace period)?

I just tried removing the final wait for the CR to be deleted and the error above appens again.

Reply:
I wonder why it was not needed before or we haven't seen it earlier.